include/lp_hash_table.h

```diff
@@ -0,0 +1,90 @@
+#ifndef DIL_LP_HASH_TABLE_H
+#define DIL_LP_HASH_TABLE_H
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <assert.h>
+
+
+template <typename T>
+struct lp_hash_table {
+private:
+    const char **keys;
+    T *values;
+    int nbuckets;
```

Comment:
Use `ssize_t` for this as the `int` can restrict us.

Reply:
Why not `size_t`. AFAIK `ssize_t` is just a signed `size_t` and `nbuckets` shouldn't be negative. Am I missing something?

Reply:
You're correct in that but you might not think the problems of unsigned integers. They have edge cases which a lot of people don't consider when writing code. A simple example:
```
for (size_t j = ...; j >= 0; --j)  // This results in an infinite loop
```